apps/managedidentity/managedidentity.go

```diff
@@ -64,7 +70,13 @@ func (c UserAssignedClientID) value() string   { return string(c) }
 func (o UserAssignedObjectID) value() string   { return string(o) }
 func (r UserAssignedResourceID) value() string { return string(r) }
 func SystemAssigned() ID {
-	return systemAssignedValue("")
+	return systemAssignedValue(systemAssignedManagedIdentity)
+}
+
+var cacheManager *storage.Manager = storage.New(oauth.New(http.DefaultClient))
```

Comment:
Yes, that is the goal - one cache per process for managed identity tokens.

Reply:
🤔 the trouble is that the cache does instance discovery to figure out environment aliases and needs an HTTP client to do that. Obviously we don't want managed identity clients to request instance metadata in any case, so we'll just disable discovery (@4gust there's an option on `authority.Info`) and the cache won't use its client. But that doesn't help with this constructor.

Separating instance discovery from the cache might be ideal but that's major surgery. I suppose we could just leave this line as-is and comment that the cache never uses the client because instance discovery is always disabled. Taking a closer look, there's a less hacky solution: the outer argument here is `*oauth.Client`, so `storage.New(nil)` is valid. It's not perfect because we would need nil checks wherever the cache uses its client, but that's better than constructing an unused client.

In any case, we need a test that fails when Client sends an unexpected request.